src/Functions/MultiMatchAnyImpl.h

```diff
@@ -245,51 +261,129 @@ struct MultiMatchAnyImpl
                 throw Exception("Failed to scan with vectorscan", ErrorCodes::HYPERSCAN_CANNOT_SCAN_TEXT);
 
             prev_haystack_offset = haystack_offsets[i];
+            prev_needles_offsets_offset = needles_offsets[i];
+            needles.clear();
         }
 #else
         /// fallback if vectorscan is not compiled
+        /// -- the code is copypasted from vectorVector() in MatchImpl.h and quite complex code ... all of it can be removed once vectorscan is
```

Comment:
This branch is called in a quite obscure edge case ... for functions mutliMatchAny() / multiMatchAnyIndex() with non-const needles on platforms where vectorscan is disabled (all non-x86 at the moment). It is a bit fragile and I had to write it from scratch using some copypasta from elsewhere because the technique used in the fallback for const needles didn't work conceptually.

I did some local testing to make sure it's okay. However, if problems in CI tests pop up, I would likely remove it altogether and throw a "not implemented" exception instead. The same is done for function multiMatchAllIndices() which is sort of related. There is a good chance that vectorscan can be enabled on non-x86 so we could throw the fallback away anyways.